Add nlp libraries survey

clachenmaier opened the pull request:
Not yet finished

**Tasks:**
- [x] Complete initial survey
- [x] Pick candidates (technically & model-wise possible and sensible to integrate)
- [ ] Compare performance (optional)

Commit:
Add OpenNLP

Commit:
Add Spacy

Commit:
Add CoreNLP

Commit:
Add Spark NLP

Commit:
Add Pattern

Commit:
Add NLP Architect

Commit:
Add Textblob

Commit:
Add CogComp NLP Pipeline

Commit:
Revise SpaCy

Commit:
Revise Standford CoreNLP

Commit:
Revise Apache OpenNlp

Commit:
Add NLP4J

Commit:
Add NLTK

sdruskat commented:
Thanks for the update. Mention me and Thomas here via `@` once you're finished :).

Commit:
Add AllenNLP

Commit:
Add repositories

Commit:
Add NER Information

Commit:
Revise Lingpipe

Commit:
Add GATE & Annie

Commit:
Remove template

Commit:
Revise AllenNLP

Commit:
Revise OpenNLP

Commit:
Revise CogComp NLP Pipeline

Commit:
Revise Architect NLP

Commit:
Revise NLTK

Commit:
Revise Pattern

Commit:
Revise Lingpipe

Commit:
Revise SpaCy

Commit:
Fix alphabetical order

Commit:
Revise SparkNLP

Commit:
Revise Stanford CoreNLP

Commit:
Revise Textblob

Commit:
Add Clear NLP

Commit:
Unify capitalization of data formats

clachenmaier commented:
I'm finished @sdruskat  @thomaskrause

Commit:
Remove wrong repositories

sdruskat committed:
Remove whitespaces and prepare PR

sdruskat committed:
Clean up whitespaces

sdruskat committed:
Fix tables for use in mdbook

sdruskat committed:
Externalize library links and add conclusion

- Links to the libraries' websites made it impossible to link to
sections internally, therefore all links have been moved to a
block below the section headings

sdruskat committed:
Fix typo

Co-authored-by: Thomas Krause <thomaskrause@users.noreply.github.com>

clachenmaier committed:
Merge pull request #1 from sdruskat/add-nlp-libraries-survey

Add executive summary
